lib/wallet/http.js

```diff
@@ -190,9 +191,23 @@ class HTTP extends Server {
         return;
       }
 
+      // SPV doesn't have blocks on disk, so rescan is ineffective
+      enforce(!this.options.node.chain.spv, 'Cannot rescan in SPV mode.' +
+        ' Try rewinding blockchain with node `reset` command.');
+
       const valid = Validator.fromRequest(req);
       const height = valid.u32('height');
 
+      // Pruned node can only rescan the blocks it has on disk
+      if (this.options.node.chain.prune) {
+        const tipHeight = await this.options.node.client.getTip();
+        const pruneDepth = this.network.block.keepBlocks;
+        const minHeight = tipHeight.height - pruneDepth;
+
+        enforce(height > minHeight,
+          'Cannot rescan past prune depth of ' + minHeight);
```

Comment:
Should we say "rescan height must be greater than minHeight"? It's a bit clearer and more direct. Bit confusing to have to think about depth when the param is height.

Reply:
👍